service/schema/SchemaService.kt

```diff
@@ -308,14 +307,16 @@ class SchemaService(
 
         var result: T? = null
         try {
-            countRunningReadAtomic.incrementAndGet()
+            countRunningReadAtomic.increment()
             openOrGetReadTx()?.let {
                 result = function(it)
             } ?: notification.userWarning(LOGGER, FAILED_TO_OPEN_READ_TX)
+        } catch (e: TypeDBDriverException) {
+            throw e
```

Comment:
We should not consume `TypeDBDriverException`s in `mayRunReadTx`, as that signals a TypeDB error that should be propagated to the caller.